apps/studio/src/server/modules/page/page.router.ts

```diff
@@ -100,6 +110,79 @@ export const pageRouter = router({
         ...siteMeta,
       }
     }),
+
+  reorderBlock: protectedProcedure
+    .input(reorderBlobSchema)
+    .mutation(async ({ input: { pageId, from, to, blocks, siteId }, ctx }) => {
+      // NOTE: we have to check against the page's content that we retrieve from db
+      // we adopt a strict check such that we allow the update iff the checksum is the same
+      const fullPage = await getFullPageById({ resourceId: pageId, siteId })
+
+      if (!fullPage?.content) {
+        // TODO: we should probably ping on call
+        throw new TRPCError({
+          code: "NOT_FOUND",
+          message:
+            "Unable to load content for the requested page, please contact Isomer Support",
+        })
+      }
+
+      const actualBlocks = fullPage.content.content
+
+      if (!isEqual(blocks, actualBlocks)) {
```

Comment:
I think for now is fine, but in future we might want to explore comparing hashes. Should see which works better with large jsons as well

Reply:
we were going to use hashes but 
1. no good shared lib between fe/be (`crypto` only backend) 
2. back of envelope calculation suggests that max size is ~15kb (3000 words each of length 5), which is ~10ms with good wifi (tested against office) 

i think there's a point to be made for hashing but we shuold do it next itme

Reply:
sgtm